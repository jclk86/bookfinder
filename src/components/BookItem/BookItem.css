li {
  margin: 0 auto;
  width: 1000px;
  list-style: none;
  border: 1px solid black;
}

li:hover {
  cursor: pointer;
}

.hidden {
  display: none;
}
