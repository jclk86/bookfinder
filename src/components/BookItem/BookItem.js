import React, { Component } from "react";
import "./BookItem.css";
import BookDetails from "../BookDetails/BookDetails";

export default class BookItem extends Component {
  constructor() {
    super();
    this.state = {
      details: "",
      isHidden: true
    };
  }

  renderDetailsOnClick = () => {
    this.setState({ isHidden: !this.state.isHidden });
  };

  render() {
    const { volumeInfo } = this.props.book;
    return (
      <li onClick={this.renderDetailsOnClick}>
        <h2>{volumeInfo.title}</h2>
        <img
          src={
            volumeInfo.imageLinks !== undefined
              ? `${volumeInfo.imageLinks.thumbnail}`
              : ""
          }
          alt="book"
        />
        <p>{volumeInfo.authors}</p>
        {!this.state.isHidden && <BookDetails book={this.props.book} />}
      </li>
    );
  }
}
