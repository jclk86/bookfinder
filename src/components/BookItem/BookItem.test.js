import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import BookItem from "./BookItem";

describe("BookItem Component", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<BookItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
