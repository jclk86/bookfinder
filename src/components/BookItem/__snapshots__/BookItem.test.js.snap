// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem Component renders without crashing 1`] = `<li />`;
